app/services/openid_connect_service.rb

```diff
@@ -0,0 +1,12 @@
+class OpenidConnectService
```

Comment:
This class is doing two unrelated things. The generation of a private key should be in its own class called something like `PrivateKeyGenerator` or `GeneratePrivateKey`. We use this same key generation code in many places, so it makes sense to move it to a dedicated class.

Also, in general, it's not recommended to add the word `Service` to a class name because it's redundant. Everything in `app/services` is a Service. Class names should be as descriptive as possible. If you remove the word `Service`, does your class name still convey what it does?